src/ucc/api/ucc.h

```diff
@@ -659,6 +660,48 @@ typedef struct ucc_oob_coll {
 
 typedef ucc_oob_coll_t ucc_context_oob_coll_t;
 typedef ucc_oob_coll_t ucc_team_oob_coll_t;
+
+/**
+ *
+ *  @ingroup UCC_CONTEXT_DT
+ */
+typedef enum  {
+    UCC_MEM_CONSTRAINT_SYMMETRIC   = UCC_BIT(0),
+    UCC_MEM_CONSTRAINT_PERSISTENT  = UCC_BIT(1),
+    UCC_MEM_CONSTRAINT_ALIGN32     = UCC_BIT(2),
+    UCC_MEM_CONSTRAINT_ALIGN64     = UCC_BIT(3),
+    UCC_MEM_CONSTRAINT_ALIGN128    = UCC_BIT(4),
```

Comment:
cuda bufferes are usually aligned to a 256bit boundary, do we want to cover this case?

Reply:
I think so. Added that constraint.